### ISSUE TITLE
NoSQL image - running in a user other than root

### ISSUE THREAD

**Issue by mikebrey** [oracle/docker-images#2604]
We have gotten a couple of requests to run the container image with a user other than root.  Is this possible?  If so, can you post the instructions on how to do this in the container file.

**Comment by dario-vega** [comment]
I am working in improving our image to address this issue

In the meantime, you can modify the Dockerfile as follow - running using a user called nosql

```bash
# Copyright (c) 2022, 2023 Oracle and/or its affiliates.
# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
#
FROM ghcr.io/graalvm/jdk:ol8-java17

LABEL org.opencontainers.image.source = "https://github.com/oracle/docker-images"

ARG KV_VERSION=22.3.32
ARG DOWNLOAD_ROOT=http://download.oracle.com/otn-pub/otn_software/nosql-database
ARG DOWNLOAD_FILE="kv-ce-${KV_VERSION}.zip"
ARG DOWNLOAD_LINK="${DOWNLOAD_ROOT}/${DOWNLOAD_FILE}"

ENV KV_PROXY_PORT 8080
ENV KV_ADMIN_PORT 5999

ENV KV_PORT 5000
ENV KV_HARANGE 5010-5020
ENV KV_SERVICERANGE 5021-5049

# hadolint ignore=DL4006
RUN curl -OLs $DOWNLOAD_LINK  && \
    jar tf $DOWNLOAD_FILE | grep "kv-$KV_VERSION/lib" > extract.libs && \
    jar xf $DOWNLOAD_FILE @extract.libs && \
    rm -f $DOWNLOAD_FILE extract.libs

WORKDIR "/app/kv-$KV_VERSION"

COPY start-kvlite.sh .
RUN chmod +x start-kvlite.sh


RUN useradd -u 1000 nosql && \
    mkdir -p /kvroot && \
    chown -R nosql /kvroot && \
    mkdir -p /shared_conf && \
    chown -R nosql /shared_conf && \
    chown -R nosql /app/kv-$KV_VERSION

VOLUME ["/kvroot"]

USER nosql

CMD ["bash", "-c", "./start-kvlite.sh"]
```

**Comment by dario-vega** [comment]
fixed by  NoSQL image - Improvements #2606
